app/src/main/java/nl/tudelft/cs4160/identitychain/main/MainActivity.kt

```diff
@@ -129,18 +127,45 @@ class MainActivity : AppCompatActivity(), CommunicationListener {
 
 
         //start listening for messages
-        communication!!.start()
+        requireNotNull(localIPAddress, { "error could not find local IP" })
+        val (server, grpc) = ChainServiceServer.createServer(kp, 8080, localIPAddress!!, dbHelper)
```

Comment:
Are we just losing the grpc reference here?

Reply:
yeah, we currently just leak the resource, it's only needed for shutting the server down when the app exits, let's make an issue for that